src/neml2/models/solid_mechanics/LinearElasticity.cxx

```diff
@@ -49,14 +49,35 @@ LinearElasticity::LinearElasticity(const ParameterSet & params)
     _rate_form(params.get<bool>("rate_form")),
     _strain(params.get<LabeledAxisAccessor>("strain").with_suffix(_rate_form ? "_rate" : "")),
     _stress(params.get<LabeledAxisAccessor>("stress").with_suffix(_rate_form ? "_rate" : "")),
-    from(declare_input_variable<SymR2>(_compliance ? _stress : _strain)),
-    to(declare_output_variable<SymR2>(_compliance ? _strain : _stress))
+    from_var(declare_input_variable<SymR2>(_compliance ? _stress : _strain)),
+    to_var(declare_output_variable<SymR2>(_compliance ? _strain : _stress))
 {
   setup();
 
   _T = transformation_tensor();
 }
 
+void
```

Comment:
I still don't get why this is needed, can't we just send the `transformation_tensor()` inputs?

Reply:
~So C is computed from E and nu at construction time. But after the model is constructed, it may be sent to other devices (other converted to other dtypes), by e.g. `model.to(torch::kCUDA);`. After that, E and nu will be on `torch::kCUDA` because they are registered parameters. But C stays on the CPU, if we do not re-evaluate the `transformation_tensor()`.~

I suddenly realize that I can just register the transformation tensor into the buffer...

Reply:
well no, actually, because if we decide to make E and nu `requires_grad=true`, we still need to recompute C.

errr, maybe I should just re-evaluate `transformation_tensor()` in every call of `set_value`.